EDMarketConnector.py

```diff
@@ -18,74 +18,69 @@
 from config import applongname, appname, appversion, appversion_nobuild, config, copyright
 
 if __name__ == "__main__":
+    journal_dir: str = config.get('journaldir') or config.default_journal_dir
+    # This must be at top level to guarantee the file handle doesn't go out
+    # of scope and get cleaned up, removing the lock with it.
+    journal_dir_lockfile = open(join(journal_dir, 'edmc-journal-lock.txt'), mode='w+', encoding='utf-8')
+
     def no_other_instance_running() -> bool:  # noqa: CCR001
```

Comment:
Does this still need to be as a closure? also that noqa can probably be removed now

Reply:
Whoops, not a closure, just in the if, but still, probably doesnt need to be in there anymore

Reply:
I'm not sure how this *is* a closure.

However, it is true that the function definitions could probably be taken out of this `__main__` clause at least.  On the other hand they *are* only used within it.